runtime/executor/src/main/java/org/apache/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -634,11 +502,12 @@ private void finalizeOutputWriters(final VertexHarness vertexHarness) {
     });
 
     // finalize OutputWriters for additional tagged children
-    vertexHarness.getWritersToAdditionalChildrenTasks().values().forEach(outputWriter -> {
-      outputWriter.close();
-
-      final Optional<Long> writtenBytes = outputWriter.getWrittenBytes();
-      writtenBytes.ifPresent(writtenBytesList::add);
+    vertexHarness.getWritersToAdditionalChildrenTasks().values().forEach(outputWriters -> {
+      outputWriters.forEach(outputWriter -> {
```

Comment:
How about closing OutputWriters in OutputCollector.close, and removing OutputWriters from VertexHarness?

i.e., remove (1) and use only (2)
(1) VertexHarness -> OutputWriters
(2) VertexHarness -> OutputCollector -> OutputWriters

Reply:
Good point! I'd like to change, but then the OutputCollector should contain `metricSender`, and I don't want to send the metrics within the `OutputCollector`. It requires some refactoring of metric sending, so I want to do it in a separate issue.

Reply:
https://issues.apache.org/jira/browse/NEMO-236